logstash-core/lib/logstash/compiler/lscl.rb

```diff
@@ -95,7 +98,11 @@ def plugin_name
 
     def expr_attributes
       # Turn attributes into a hash map
-      self.attributes.recursive_select(Attribute).map(&:expr).map {|k,v|
+      self.attributes.recursive_select(Attribute).map {|attribute|
+        attribute.source_file = source_file
```

Comment:
~~seems like the attr writes are only internal ... not sure - feel free to ignore if that is not the case~~